api-docs/guide/adding-a-custom-udf.rst

```diff
@@ -0,0 +1,88 @@
+.. _guide-adding-a-custom-udf:
+
+Adding a custom UDF 
+===================
+
+    This section is an overview of how you can create and execute a custom UDF. 
+
+Introduction to UDFs
+=====================
+
+    * UDFs are user-defined functions, used to achieve a specific functionality on any video loaded to EVA. 
+    * As the name suggests, these functions are defined and implemented by users. Once the UDFs are registered, you can 'call' these functions in your queries. 
+
+Implementing your own UDF
+=========================
+
+    1. Create a new file under `src/udfs/` and give it a descriptive name. eg: `fastrcnn_object_detector.py`, `midas_depth_estimator.py`. 
```

Comment:
Do you think we should have these udfs outside the src/ directory? We can create a /udf directory to avoid readers from getting confused.

Reply:
I think this looks fine as it is? All core code components going into src/ makes sense for me. Prof. @jarulraj @suryatejreddy can comment.

Reply:
I think it should be outside `src/`. Just because we want to show that the UDF is not part of the EVA Codebase and is something external that users can create. However, if we want to package some UDF's as part of the EVA module then they should be inside src, so depending on that, we should take the call I guess. @gaurav274 @Anirudh58 @jarulraj

Reply:
Yes, I agree with Surya. Users should add udfs outside the `src/ ` directory.

Reply:
Makes sense. I'll change accordingly.